### DISCUSSION TITLE
Header widget improvements

### DISCUSSION THREAD

**Discussion by davetapley** [Textualize/textual#2890] [Ideas]
Will you accept a PR to make the icon optional?
https://github.com/Textualize/textual/blob/a35e92b9fce7237fbc7db9534b0b888b4cb97c53/src/textual/widgets/_header.py#L153-L155

---

Alternatively perhaps the icon should have some docs to show how to set it. I see ⬇️ but I'm not sure what how you'd change that?
https://github.com/Textualize/textual/blob/a35e92b9fce7237fbc7db9534b0b888b4cb97c53/src/textual/widgets/_header.py#L27

Then it could be allowed to be `None` (I'd argue that should be the default) and that would be how you turn it off and on?

---

Similarly I'm not sure about the default 'click to make it tall' semantics?
It seems like that should be something you should have to opt in to?

e.g. if I have custom content `hello there`:

<img width="380" alt="Screen Shot 2023-07-04 at 11 38 16" src="https://github.com/Textualize/textual/assets/48232/897a3e30-8202-481e-b2c2-14b723597cc9">

But then if someone clicks on it the see the app name below and center aligned? 😬 

<img width="557" alt="Screen Shot 2023-07-04 at 11 38 08" src="https://github.com/Textualize/textual/assets/48232/98c41d8c-1936-4fba-9357-3437093e8350">

---

Also less critical, but I feel like the default header and footer background colors should be the same?

https://github.com/Textualize/textual/blob/a35e92b9fce7237fbc7db9534b0b888b4cb97c53/src/textual/widgets/_header.py#L115-L119

https://github.com/Textualize/textual/blob/a35e92b9fce7237fbc7db9534b0b888b4cb97c53/src/textual/widgets/_footer.py#L34-L36

**Comment by willmcgugan** [comment]
The Header was one of the first widget we did, and it could certainly use some more work.

I think you can set the icon to empty string, or maybe a space to disable it.

**Reply by davetapley** [reply]
@willmcgugan thanks, would this be the preferred way?

```py
    def on_mount(self) -> None:
        header_icon = self.query_one(HeaderIcon)
        header_icon.icon = ''
```

**Reply by willmcgugan** [reply]
Should work. Although I'd probably do it in `compose`
